tools/mlir_to_verilog_main.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+import dataclasses
+import logging
+import os
+import sys
+from typing import Dict
+
+from magma.backend.mlir.mlir_to_verilog import mlir_to_verilog, MlirToVerilogOpts
+from magma.common import slice_opts
+
+
+logging.basicConfig(level=os.environ.get("LOGLEVEL", "WARNING").upper())
+
+
+def _field_to_argument_params(field: dataclasses.Field) -> Dict:
+    if field.default_factory is not dataclasses.MISSING:
+        raise TypeError(field)
+    params = {
+        "required": field.default is dataclasses.MISSING,
+    }
+    if field.type is bool and not params["required"] and not field.default:
+        params.update({"action": "store_true"})
```

Comment:
```suggestion
        params["action"] = "store_true"
```

Reply:
Could just use key assignment for these?

https://stackoverflow.com/questions/15456158/python-dict-update-vs-subscript-to-add-a-single-key-value-pair